New slide creation

skatulich opened the pull request.

kemaric committed:
Update issue templates

kemaric committed:
Merge pull request #6 from Swen670Grp1/kemaric-patch-1

Update issue templates

skatulich committed:
Initial checkin

Adds basic functionality to add new slides to presentation through Javascript.

skatulich committed:
Update to slide styling

skatulich committed:
Updates to add 2nd slide

includes basic css

skatulich committed:
Merge branch 'master' into NewSlideCreation

skatulich committed:
Add read configuration file

skatulich committed:
New Slide creation function

Function creates new slide based on header, body, and footer text/data.

skatulich committed:
Merge branch 'develop' into NewSlideCreation

kemaric commented on src/app/css/reveal.css:
```diff
@@ -396,8 +396,8 @@ body {
-/*********************************************
- * SLIDE NUMBER
+/*********************************************
+ * SLIDE NUMBER
```
I have no idea why these are showing up as changes. I dont see the difference.

kemaric commented on src/app/index.html:
```diff
@@ -14,167 +14,73 @@
-  <script src="https://cdn.jsdelivr.net/npm/ua-parser-js@0/dist/ua-parser.min.js"></script>
-
-  <script src="js/cycaller.js"></script>
-  <script src="js/slides/session.js"></script>
```
Any reason why cycaller.js, session.js, slidesModel.js were removed from Index.html? Without them, those scripts arent ran.

kouissar replied:
Not sure if these tell you the whole story. I checked out the branch and I see these included in index.html

kemaric replied:
@kouissar at the time of my comment, those scripts were not included. Commits were pushed to include them.

kouissar replied:
Possible! :)

kemaric commented on src/app/js/main.js:
```diff
@@ -6,6 +6,25 @@ function toggleLog() {
+  loadJSON(function(response) {
+    var configuration = JSON.parse(response);
+  });
+  addResponse(slide.id, { 'appVersion': window.navigator['appVersion'] })
```
This is missing a semi colon at the end.

kemaric replied:
I also see two `init` function. In your new one, you don't create the TestSession at all anymore.

kemaric reviewed (changes requested):
This branch has a js error in the main.js file beasue of the two init function. The original has the creation of the TestSession, while the new one does not. Can you resolve this?

skatulich committed:
Updates per code review

Added missing js files.
Updated main.js
 - included loadJSON
 - removed extran init
 - added semi colons to other lines that appeared to be missing them.

skatulich committed:
Updates to index.html

skatulich committed:
Updates for semi colons

skatulich commented:
All updates have been made except for the Session.js file does not seem to be valid.  There are lots of semi colons missing throughout the javascript.  I added the develop JS files and now have an error for UAParser not being defined and "style" not being defined in main.js.

skatulich committed:
Adding semi colons

skatulich committed:
Remove unneeded script

skatulich committed:
resolved UAParser error

kouissar commented on src/app/index.html:
```diff
@@ -14,167 +14,77 @@
-        <textarea id="log"></textarea>
-      </div>
-    </div>
-  <div id="reveal-container" style="width: 100%; height: 100%">
```
your removal of reveal container is causing this error at this line: revealContainer.style.height = '100%'
Uncaught TypeError: Cannot read property 'style' of null
    at toggle_tests (main.js:136)
    at call (main.js:413)
    at setTimeout (main.js:514)

kouissar commented on src/app/index.html:
```diff
@@ -14,167 +14,77 @@
+						</div>
+							<div class="column right">
+								<div>
+									<label class="labelText">Tester Name:</label>
```
Let's move tester name field to the bottom of welcome text, it doesn't look good to me. Also log button needs to have a better position on the screen, I believe initial position was better.

kouissar commented on src/app/index.html:
```diff
@@ -14,167 +14,77 @@
-        <input type="button" id="jiraBtn2" onClick="attach()" value="Attach"/> -->
-
-      </div>
-      <div id="log-container" style="display:none">
```
Steve, How do you plan to expose runtime errors on the UI? do you plan to have an error container to display caught errors, or use buildslide html to render the errors?

kemaric replied:
I think that's outside the scope of this. But that being said, it can be as simplying as using some bootstrap alert/toaster in the application.

kemaric reviewed (changes requested):
When running the application, I get the following error.
![image](https://user-images.githubusercontent.com/1737082/48685732-c6020900-eb85-11e8-81d3-ecc4823e466b.png)
Can you address that.

skatulich committed:
Updates for merge

Updates index.html to include reveal-container.
Updates main.css to left align log button.

skatulich committed:
Final update for code review

Final update to include div id=tests

skatulich committed:
Removed unnecessary code

skatulich committed:
Final updates

skatulich commented:
Can you please update your reviews so this can be merged in?

kouissar reviewed (dismissed):
This is OK for now, we will be able to see how things shape up once other core functionality is merged and test slides are implemented.

skatulich committed:
Merge branch 'develop' into NewSlideCreation

skatulich commented:
Please accept so I can merge.

kemaric reviewed (changes requested):
I can't accept this as it's broken when the code is ran. I think you didn't merge this branch properly after your last PR went in.

kouissar reviewed (dismissed):
Approving for now, we should see how the app looks once everything is hooked.

skatulich committed:
Merge branch 'develop' into NewSlideCreation

skatulich committed:
Updates to merge with Development

kemaric committed:
Disabling the textarea for the log.

kouissar commented:
I tested it however I feel this is a regression vs the working version we have in develop. As I shared with Kemari, It is too high of a risk to introduce such big of a change that might render the app unusable at this late stage. I suggest we finalize develop and test it so at least we have a working model with few working and helpful features that can be augmented by Cytoscape team or Spring class.  This is great work though, I would keep the configuration changes but cancel the the UI or design changes.
